### PR TITLE
Raise error for colors with leading #

### PR DESCRIPTION
**Pull request by tomprats**
Saw this [issue](https://github.com/prawnpdf/prawn/issues/807) and thought I could help! Really appreciate all the work you guys do!

### CODE REVIEW

**Comment by packetmonkey** [comment]
Nice! Thanks for the PR!

First off please add an entry to the change log so we can doc the updated behavior.

Also I'm not sure that checking for the length of the string will be enough, as someone could say pass `EEFFGG`. What do you think about updating the logic to
```ruby
when String
  if color =~ /\A[0-F]{6}\z/i
    :RGB
  else
    raise ArgumentError, "Unknown type of color: #{color.inspect}"
  end
```

Thanks again!

**Comment by tomprats** [comment]
Good call! I wasn't thinking about the actual string content, I'll update both of those

**Comment by tomprats** [comment]
I'm just going to close this because it's probably no longer valid

**Comment by packetmonkey** [comment]
Hey I missed that you updated this branch, if you are still interested in getting it in I'm open to merging.

Thanks!

**Comment by tomprats** [comment]
@packetmonkey Thanks! Should be good to go now

File: manual/graphics/color.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@
 Prawn::ManualBuilder::Example.generate(filename) do
   stroke_axis
 
-  # Fill with Yellow using RGB
+  # Fill with Yellow using RGB (Unlink css, there is no leading #)
```
Comment (crazymykl) [inline manual/graphics/color.rb]: Unlike.

File: lib/prawn/graphics/color.rb [hunk 2]
```diff
@@ -94,7 +94,11 @@ def process_color(*color)
       def color_type(color)
         case color
         when String
-          :RGB
+          if color =~ /\A[0-F]{6}\z/i
```
Comment (pointlessone) [inline lib/prawn/graphics/color.rb]: There are a few characters that we probably don't want to match between `'9'` and `'A'`. I'd prefer this regex to be something more specific. For example `/\A[0-9a-f]{6}\z/i`.

**Comment by pointlessone** [comment]
@tomprats If you're steal interested, please address my comment and rebase your branch on master. Thank you.

**Comment by pointlessone** [comment]
@tomprats Than you for your contribution. I've rebased this PR on master and merged. GitHub didn't pick it up so I will close it now.

**Comment by tomprats** [comment]
@pointlessone Thanks! Sorry I never got around to making the extra changes

**Pull request closed without merging**